### PR TITLE
BifurcationKit test

### PR DESCRIPTION
**Pull request by vyudu**
Wrote a test for the dependent parameters for BK from #1095, that can be uncommented once that works

### CODE REVIEW

**Comment by isaacsas** [comment]
Let’s merge this once MTK releases with your update since then v.4 should work here. 

You can use `@test_broken` for now as that would let us know when the test starts working.

**Comment by isaacsas** [comment]
@vyudu if you want to merge master into this the SciMLBase cap will hopefully allow it to run through to completion.

**Comment by isaacsas** [comment]
@vyudu the doc folder Project.toml needs updating too.

File: Project.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@ CatalystHomotopyContinuationExtension = "HomotopyContinuation"
 # CatalystStructuralIdentifiabilityExtension = "StructuralIdentifiability"
 
 [compat]
-BifurcationKit = "0.3"
+BifurcationKit = "0.4.3"
```
Comment (isaacsas) [inline Project.toml:45]:
```suggestion
BifurcationKit = "0.4.4"
```

File: docs/Project.toml [hunk 2]
```diff
@@ -45,7 +45,7 @@ Symbolics = "0c5d862f-8b57-4792-8d23-62f2024744c7"
 
 [compat]
 BenchmarkTools = "1.5"
-BifurcationKit = "0.3.4"
+BifurcationKit = "0.4"
```
Comment (isaacsas) [inline docs/Project.toml:48]:
```suggestion
BifurcationKit = "0.4.4"
```

**Comment by isaacsas** [comment]
OK, seems there are now new intialization related errors. That never happened under the old extension so I guess this is something broken by recent updates to the MTK-BifurcationKit extension (or initialization related changes). 

If someone wants to take a look at what is going on feel free. I think for now I will just plan to disable this extension like we've disabled SI as I don't have the time or capacity to handle these breakages.

**Comment by ChrisRackauckas** [comment]
> That never happened under the old extension so I guess this is something broken by recent updates to the MTK-BifurcationKit extension (or initialization related changes).

BifurcationKit is broken due to Krylov.jl things, completely unrelated to anything MTK?

**Comment by isaacsas** [comment]
And this CI failure is something symbolic related: https://github.com/SciML/Catalyst.jl/actions/runs/11859907434/job/33053954257?pr=1101#step:6:1289

**Comment by isaacsas** [comment]
@vyudu I dropped your new test. It had a parameter that was not used and no value specified which was causing it to fail CI, and the ODESystem test part seems unrelated to BifurcationKit (so while a good test should probably be placed in a different location).

If you want to followup if/when this gets merged and modify those tests feel free (I've left them commented out for now).

Hopefully adding a parent field to `ReactionSystem`s will fix the symbolic indexing issues.

**Comment by vyudu** [comment]
Ahh, I see, leaving `ratechange` in was sloppy sorry. Will revisit this after merge

**Pull request merged**